scene/resources/curve.h

```diff
@@ -258,6 +258,7 @@ class Curve3D : public Resource {
 	};
 
 	Vector<Point> points;
+	mutable Vector<size_t> points_in_cache; // Index in the backed caches corresponding to control points.
```

Comment:
I assume `points_in_cache` and `get_point_baked_posture()` should be implemented on the EditorPlugin to begin with. You should add the `"curve_baked"` signal to `Curve::_bake()` and calculate it on the EditorPlugin side.

Reply:
Since each point can already be retrieved individually, you can implement this in EditorPlugin by putting `compose_posture()` in public just to reuse it. There is no need to put caches in the object that are not actually needed at runtime.

Reply:
> You must avoid to put cache and methods in the object which are not actually needed in game.

Agreed. But here `points_in_cache` and `get_point_baked_posture()` are needed in game. Say I am developing some game like city skyline. I want in game editor for express roads which are represented by `Curve3D`, then I need to access posture of the control points(not baked points), then I need call `get_point_baked_posture()`.

Reply:
I see, then I think you can just remove the ifdef restriction at the end of this explanation and expose it to gdscript.

https://github.com/godotengine/godot/pull/80329#discussion_r1290160880

> It would be most memory saving to create this cache array as a local variable in Curve3D::_bake(), and pass it to Path3DEditor via the signal argument like emit_signal("curve_baked", point_map_array), and let the Editor cache it by duplicating it. Then, you would need to run _bake() each time you select the curve. Also, the array calculation and signal should only be fired with #ifdef TOOLS_ENABLED.

Reply:
since the main difference of opinions is discussed in other thread. I close this for readability.